packages/ui/atomic/create-atomic-component/index.js

```diff
@@ -125,39 +141,56 @@ const __dirname = dirname(fileURLToPath(import.meta.url));
 const templateRelativeDir = 'template';
 const templateDirPath = resolve(__dirname, templateRelativeDir);
 
-cpSync(templateDirPath, cwd(), {
-  recursive: true,
-});
-
-let componentName = process.argv[2];
-if (componentName) {
-  ensureComponentValidity(componentName);
-  const transformers = [
-    {
-      srcPath: 'src/components/sample-component',
-      destPath: `src/components/${componentName}`,
-    },
-    {
-      srcPath: `src/components/${componentName}/src/sample-component.tsx`,
-      destPath: `src/components/${componentName}/src/${componentName}.tsx`,
-      transform: (text) =>
-        text
-          .replaceAll(/sample-component/g, componentName)
-          .replaceAll(/SampleComponent/g, camelize(componentName)),
-    },
-    {
-      srcPath: `src/components/${componentName}/src/sample-component.css`,
-      destPath: `src/components/${componentName}/src/${componentName}.css`,
-    },
-    {
-      srcPath: `src/components/${componentName}/package.json`,
-      destPath: `src/components/${componentName}/package.json`,
-      transform: (text) =>
-        text.replaceAll(/(@coveo\/)?sample-component/g, componentName),
-    },
-  ];
-
-  transform(transformers);
-}
+const main = () => {
```

Comment:
IMO, the 'let's put everything in a function and call it later down' is not a nice pattern IMO, especially since top-level awaits works in ESM (and this project is an ESM).

It adds an unnecessary level of tabulation.